python-main.js

```diff
@@ -1730,6 +1737,12 @@ function web_editor(config) {
             });
         }
         $("#modal-msg-links").html((modalLinks).join(' | '));
+        focusModal("#modal-msg-overlay");
+        $(document).keydown(function(e) {
+            if (e.which == 27) {
+                modalMsgClose();
+           }
+       });
```

Comment:
Every time the modalMsg function is executed (e.g. every time the connect button is clicked in firefox/safari)  it adds a new listener to the escape key that is never removed. So if I open that modal 3 times, every time the escape key is pressed (like when we want to move from the text editor to the buttons) it will try to execute modalMsgClose() three consecutive times.

Reply:
Adding `$(document).off("keydown");` to `modalMsgClose()` and `flashErrorClose()` fixes this, is that an appropriate fix?

Reply:
Will this will remove all keydown events? That could affect other features, like escaping from the text editor to the buttons.

Reply:
It does, but we use `keyup` to escape the editor to the download button. Is there a way to turn off specific events?

Reply:
Yes, you can remove a function from an event handler. The only issue is that the function has to be accessible/in-scope from the function that sets the event and the one that removes it.

Reply:
I think this is fixed by adding the event to the div element rather than the document in https://github.com/bbcmicrobit/PythonEditor/pull/310/commits/129bfa6abd4488dec28873a671f661cfd41e86ce have tested across browsers